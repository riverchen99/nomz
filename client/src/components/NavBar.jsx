import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../images/nomz_logo.png';
import HamburgerMenu from '../images/hamburger_menu.png';
import { StyledNavBar, LogoImage, User, HamburgerImage, Menu, Wrapper, Item } from './StyledNavBar';

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false,
    }
  }
  render() {
    return (
      <Wrapper>
        <StyledNavBar>
          <LogoImage src={Logo} />
          <User>{this.props.userName}</User>
          <HamburgerImage src={HamburgerMenu} onClick={() => this.setState(prevState => ({ expanded: !prevState.expanded })) }></HamburgerImage>
        </StyledNavBar>
        <Menu expanded={this.state.expanded} >
          <Item><Link to={{ pathname: '/profile'}} style={{ textDecoration: 'none', color: '#888888' }}>Profile</Link></Item>
          <Item><Link to={{ pathname: '/recommend'}} style={{ textDecoration: 'none', color: '#888888' }}>Top Rated</Link></Item>
          <Item><Link to={{ pathname: '/allmenus' }} style={{ textDecoration: 'none',  color: '#888888' }}>Menus</Link></Item>
          <Item><a href="http://localhost:8080/auth/logout">Logout</a></Item>
        </Menu>
      </Wrapper>
    )
  }
}

export default NavBar;