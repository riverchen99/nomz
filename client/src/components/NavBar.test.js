import React from 'react';
import { shallow, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import NavBar from './NavBar';

configure({ adapter: new Adapter() });

describe('Navbar component', () => {
  it('should render correctly', () => {
    const component = shallow(<MemoryRouter><NavBar userName="Guest" /></MemoryRouter>);
    expect(component).toMatchSnapshot();
  });
});
