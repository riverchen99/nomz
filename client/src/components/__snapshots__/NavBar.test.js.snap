// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should keep track of logged in state 1`] = `ReactWrapper {}`;

exports[`Navbar component should render correctly 1`] = `ShallowWrapper {}`;
